aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -589,7 +589,20 @@ void logit_backward_kernel(TensorIterator& iter, Scalar eps_scalar) {
 }
 
 void tanh_backward_kernel(TensorIterator& iter) {
-  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(iter.dtype(), "tanh_backward_cpu", [&]() {
+  if (isComplexType(iter.dtype())) {
+    AT_DISPATCH_COMPLEX_TYPES(iter.dtype(), "tanh_backward_cpu", [&]() {
```

Comment:
Why is this one separated? Isn't the conj a no-op for reals? Or is the problem that it returns a complex?

Reply:
`std::conj` returns a complex number for real numbers as well https://en.cppreference.com/w/cpp/numeric/complex/conj. so because it's not a no op for real numbers, I split up dispatch for complex and non-complex dtypes.